src/components/SettlementButton.tsx

```diff
@@ -270,7 +274,12 @@ function SettlementButton({
             return;
         }
 
-        if (iouPaymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY || iouPaymentType === CONST.IOU.PAYMENT_TYPE.VBBA) {
+        if (
+            iouPaymentType === CONST.IOU.PAYMENT_TYPE.VBBA ||
+            iouPaymentType === CONST.PAYMENT_METHODS.BUSINESS_BANK_ACCOUNT ||
+            iouPaymentType === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT ||
+            iouPaymentType === CONST.PAYMENT_METHODS.DEBIT_CARD
+        ) {
             triggerKYCFlow(event, iouPaymentType);
             BankAccounts.setPersonalBankAccountContinueKYCOnSuccess(ROUTES.ENABLE_PAYMENTS);
```

Comment:
Looks like this logic caused the following issue:
- https://github.com/Expensify/App/issues/49523

Because on unvalidated (gmail) accounts, when a user tries to Pay someone with Expensify, the RHP opens with `this page cannot be found` screen because of [this code](https://github.com/Expensify/App/blob/d009d8ed4830c7d0562c524c991b8a688e2b4869/src/pages/AddPersonalBankAccountPage.tsx#L70) from `AddPersonalBankAccountPage`. We fixed by routing the user to verify their account before `Pay with Expensify` in case it's unverified.